.github/scripts/upload_tests_results.py

```diff
@@ -0,0 +1,342 @@
+        except Exception as e:
+            print(f"Error executing query {e}, sql:\n{sql}", file=sys.stderr)
+            raise e
+
+
+def prepare_and_upload_tests(pool, path, results, batch_size):
+
+    total_records = len(results)
+    if total_records == 0:
+        print("Error:Object stored test results is empty, check test results artifacts")
+        return 1
+
+    with ThreadPoolExecutor(max_workers=10) as executor:
+
+        for start in range(0, total_records, batch_size):
+            futures = []
+            end = min(start + batch_size, total_records)
+            batch = results[start:end]
+
+            sql = f"""--!syntax_v1
+            UPSERT INTO `{path}` 
+            (build_type,
+            job_name,
+            job_id,
+            commit,
+            branch,
+            pull, 
+            run_time, 
+            test_id, 
+            suite_name, 
+            raw_test_name, 
+            file, 
+            test_name, 
+            fixture, 
+            time, 
+            status,
+            status_description,
+            log,
+            logsdir,
+            stderr,
+            stdout,
+            owners) VALUES"""
+            values = []
+            for index, result in enumerate(batch):
+
+                values.append(
+                    f"""
+                ("{result['build_type']}", 
+                "{result['job_name']}", 
+                {result['job_id']}, 
+                "{result['commit']}",
+                "{result['branch']}",
+                "{result['pull']}", 
+                DateTime::FromSeconds({result['run_time']}),
+                "{result['pull']}_{result['run_time']}_{start+index}",
+                "{result['suite_name']}",
+                "{result['raw_test_name']}",
+                "{result['file']}",
+                "{result['test_name']}",
+                "{result['fixture']}",
+                {result['time']},
+                "{result['status']}",
+                "{result['status_description']}",
+                "{result['log']}",
+                "{result['logsdir']}",
+                "{result['stderr']}",
+                "{result['stdout']}",
+                "{result['owners']}")
+                """
+                )
+            sql += ", ".join(values) + ";"
+
+            futures.append(executor.submit(upload_results, pool, sql))
+
+        for future in as_completed(futures):
+            future.result()  # Raise exception if occurred
+
+
+def create_pool(endpoint, database):
+    driver_config = ydb.DriverConfig(
+        endpoint,
+        database,
+        credentials=ydb.credentials_from_env_variables()
+    )
+
+    driver = ydb.Driver(driver_config)
+    driver.wait(fail_fast=True, timeout=5)
+    return ydb.SessionPool(driver)
+
+
+def main():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--result-file', action='store', dest="result_file", required=True, help='XML with results of tests')
```

Comment:
`--test-results` is better, because someone may think, that `--result-file` is a result of current script (not an input file)